tools/lib/provision_inner.py

```diff
@@ -93,8 +93,12 @@ def write_command(command: str) -> None:
             with open(shell_profile_path, "w") as shell_profile_file:
                 shell_profile_file.writelines(command + "\n")
 
-    source_activate_command = "source " + os.path.join(VENV_PATH, "bin", "activate")
-    write_command(source_activate_command)
+    host_machine_wsl = os.path.exists("/proc/sys/fs/binfmt_misc/WSLInterop")
+    host_machine_native_linux = os.path.exists("/etc/os-release")
+    if not host_machine_wsl or not host_machine_native_linux:
```

Comment:
Can you explain how this equates to `non-vagrant/docker/droplet`?

Reply:
Yes, of course! So, I'm evaluating the host machine type based on the unique file system path associated with each machine.
The path "/proc/sys/fs/binfmt_misc/WSLInterop" is unique to WSL machines and not found in Vagrant, Docker, or Droplet containers. Similarly, the path "/etc/os-release" is a common file system path in native Linux containers for identifying the specific distribution and version.
For those machines that have one of these paths in their file system, the activation command will not get written in those respective bash profiles.

Reply:
`"/etc/os-release"` can also be found in vagrant/docker/droplets right?

Reply:
I think you're right, It could be found in some Vagrant/Docker/Droplet setup. So I think I need to change the way it's evaluating machine type for native Linux machines. My question is, other than that, is my way of approaching this issue is correct? What do you think?

Reply:
What if I alter my logic a little bit, instead of checking for paths that are specific to native Linux machines, I will check for paths that are specific to vagrant/docker/droplet containers. For example these paths  `</var/run/docker.sock>`,  `</vagrant>` and `<etc/digitalocean>` are specific paths for docker, vagrant and droplet machines respectively. I can modify the logic to check for this paths.